### PR TITLE
SLI-686 Resolve the server branch based on local repo

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

**Comment by damien-urruty-sonarsource** [comment]
@jblievremont maybe better to do the review commit by commit

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -141,6 +141,7 @@
     <!-- please see http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/plugin_compatibility.html
          on how to target different products -->
     <depends>com.intellij.modules.lang</depends>
+    <depends>Git4Idea</depends>
```
Comment (jblievremont) [inline src/main/resources/META-INF/plugin.xml:144]: Shall we make it optional already in this PR, or later when working on the events?

File: src/main/java/org/sonarlint/intellij/core/EngineManager.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.sonarlint.intellij.exception.InvalidBindingException;
+import org.sonarlint.intellij.notifications.SonarLintProjectNotifications;
+import org.sonarsource.sonarlint.core.client.api.common.SonarLintEngine;
+import org.sonarsource.sonarlint.core.client.api.connected.ConnectedSonarLintEngine;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneSonarLintEngine;
+
+public interface EngineManager {
```
Comment (jblievremont) [inline src/main/java/org/sonarlint/intellij/core/EngineManager.java:30]: Nice refactoring 👍🏻

**Review by jblievremont — APPROVED** [review]
LGTM! I'll merge both branches (yours first, then mine rebased) and we can work together on the remaining points.

**Pull request merged**